ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -821,4 +838,21 @@ public class RNTrackPlayer: RCTEventEmitter, AudioSessionControllerDelegate {
 
         sendEvent(withName: "playback-track-changed", body: dictionary)
     }
+
+    func handleAudioPlayerSecondElapse(seconds: Double) {
+        // because you cannot prevent the `event.secondElapse` from firing
+        // do not emit an event if `progressUpdateEventInterval` is nil
+        if progressUpdateEventInterval == nil { return }
+
+        let track = player.items[player.currentIndex] as! Track
+        sendEvent(
+            withName: "playback-progress-updated",
+            body: [
+                "position": player.currentTime,
+                "duration": player.duration,
+                "buffered": player.bufferedPosition,
+                "track": track.toObject(),
```

Comment:
@dcvz I figured we'd have a discussion about this. My rationale here was that we should pass back some data which identified the track the progress update referred to. If it's not passed then the first thing everyone is going to have to do is identify the track somehow. The most convenient way from my perspective is to pass the Track object as people likely have some sort of ID on it (even though it's not required, but allowed for based on the Track interface). The other option would be to pass back the index of the track which is probably workable but at least from my perspective isn't as convenient.

Reply:
Yeah.. I guess this event won't be so helpful without the context of the track you're getting an update for.

_What would be the alternative?_
You'd have to make a call to `getCurrentTrack()` to get the index and `getTrack()` to get the object.

Is this so bad though?

-----

The two big use cases were:
1. Sleep timers (for this you probably don't really need the current track info)
2. Analytics (people might want the current track data?)

My biggest concern is just that this `track` object could be any payload the user sent in as the track.. so we're potentially sending variable size objects that the user may or may not want.

Reply:
My spidey-sense was tingling for that same reason. How about this:

If a _track.id_ is set we send it alone back in the payload, if not we send nothing.If people care about it then they can set an id on their track. It's more opinionated but maybe that's a good thing and it's a fixed size.

The thing I worry about with fetching data based on indexes in JS is latent events towards the end of a track playback. You could end up setting an incorrect progress for a track based on timing without a concrete way to ID the event.

Reply:
@dcvz want to make sure this is showing up for you (since we've been going back and forth today)

Reply:
This might be a good first step! We may have some time during the RC's to have others who need it test it and give feedback. I'm ok with proceeding with this

Reply:
I guess we're tailoring the implementation too much towards knowing that people will use of for analytics as opposed to sending the track index and allowing them to query for the track object if they want.

Reply:
So are you saying you'd rather send an `index` instead of a `trackId`? I suppose it's relatively unlikely that the index will change during an even processing.

Reply:
Yeah.. thinking more about it I think we should return the index. Everything in this library is index based, so I think that makes sense.

Users can still grab the track object if need be by calling `getTrack`. Returning an identifier is too tailored towards analytics and forces people to use an `id` if they want to take advantage of this.

Now they can use any keys they want and only get the object if they want to for their purposes.

Reply:
I'm fine with that since it seems it would be highly unlikely for the position to change. However, if there are valid cases where the position in the queue can change during these events (for instance someone inserting at a specific index, which isn't something I do) it may be the case that we _also_ need to return a `trackId` too.